tokenizers/src/tokenizer/encoding.rs

```diff
@@ -208,13 +208,27 @@ impl Encoding {
 
     /// Truncate the current `Encoding`.
     ///
-    /// Panic if `stride >= max_len` or `max_len == 0`.
+    /// Panic if `stride >= max_len`
     pub fn truncate(&mut self, max_len: usize, stride: usize) {
         if max_len >= self.ids.len() {
             return;
         }
-        // We only truncate if max_len > 0, it makes no sense otherwise
-        assert!(max_len > 0);
+
+        if max_len == 0 {
+            let o = Encoding {
```

Comment:
I'm not sure if this is the correct/best way to do this. I'm not very familiar with Rust, so was mostly trying things and following the guidance from the compiler.